### PR TITLE
fix: fe liquidity txs not showing issues

### LINKED ISSUE #330: AddRemoveLiquidityTrans in transaction history did not work
**Describe the bug**
Adding liquidity or removing liquidity txs in transaction history dont show up after the txs succeeded
addRemoveLiquidityTrans in transaction history seemed not pulling the data

**To Reproduce**

1. Go to LP
2. Add liquidity or remove liquidity
3. Go to position
4. No lp history showing

**Expected behavior**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by yichiehliu**
# Task:

## Description

1. add useLiquidityTxHistory hook, handling lp txs pulling and calculations

Fixes #330  (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Locally with address having lp txs history

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

### CODE REVIEW

**Review by KMKoushik — COMMENTED** [review]
Can you explain more about the fix? High level it looks like you moved the function and added a subscription. Does subscription fix the issue? 

@EstherSeyi Was mentioning something about a subscription is not working in some cases, can you sync with her and let me know if that's not the problem  here?

File: packages/frontend/src/state/transactions/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          txId: transaction.transaction.id,
+          ethPriceAtDeposit: BIG_ZERO,
+        }
+
+        const squeethDepositedAmount = new BigNumber(
+          isWethToken0 ? transaction.depositedToken1 : transaction.depositedToken0,
+        )
+        const ethDepositedAmount = new BigNumber(
+          isWethToken0 ? transaction.depositedToken0 : transaction.depositedToken1,
+        )
```
Comment (adam5909099) [inline packages/frontend/src/state/transactions/hooks.ts:72]: If you define squeethDepositedAmount and ethDepositedAmount in the above transaction details, you can reuse it when defining transactionDetails.

Comment (adam5909099) [inline packages/frontend/src/state/transactions/hooks.ts:72]: If you define squeethDepositedAmount and ethDepositedAmount in the above transaction details, you can reuse it when defining transactionDetails.

Comment (yichiehliu) [reply packages/frontend/src/state/transactions/hooks.ts:72]: cuz they r only used in liquidity txs, not sure if we put them into txs details

**Comment by yichiehliu** [comment]
> Can you explain more about the fix? High level it looks like you moved the function and added a subscription. Does subscription fix the issue?
> 
> @EstherSeyi Was mentioning something about a subscription is not working in some cases, can you sync with her and let me know if that's not the problem here?

yh subscription fixed the issue, but it's a bit hard to debug with subscription cuz we dont know what it does underneath the hood, but pulling may not be good for performance, I am trying to come up with better solutions with either subscription or pulling

Would love to know if any thoughts/suggestions @KMKoushik @adam5909099 @EstherSeyi @block902

**Comment by EstherSeyi** [comment]
It doesn't seem to be working for me. In the screenshot below, I just opened an lp position, but it doesn't show up in the transactions.

Screenshot 2022-04-25 at 09 38 13

**Comment by yichiehliu** [comment]
> It doesn't seem to be working for me. In the screenshot below, I just opened an lp position, but it doesn't show up in the transactions.
> 
> Screenshot 2022-04-25 at 09 38 13

@EstherSeyi  Could u try this link?
https://continuouscall-q2bo64a51-opynfinance.vercel.app/lp

**Comment by EstherSeyi** [comment]
> continuouscall-q2bo64a51-opynfinance.vercel.app

I'm experiencing the same thing here too

File: packages/frontend/src/state/transactions/hooks.ts (hunk 1 above)
Comment (KMKoushik) [reply packages/frontend/src/state/transactions/hooks.ts:72]: @yichiehliu Can you address this

Comment (yichiehliu) [reply packages/frontend/src/state/transactions/hooks.ts:72]: @adam5909099 they r only used in liquidity txs, dont see why we move them into transactionDetails

**Review by KMKoushik — COMMENTED** [review]
Is there a way to add some tests so that it's not happening again?

**Comment by KMKoushik** [comment]
Also please focus on 3 start issues before working on this.

**Comment by yichiehliu** [comment]
> Is there a way to add some tests so that it's not happening again?

yh, was working on the tests, will be focusing on 3 star issues first and come back here after those fixes!

**Pull request closed without merging**